### PR TITLE
Send coords to player on death

### LINKED ISSUE #2896: Coords in Death Message
### Feature request

**Feature description**
I would like an option to include death coords in death messages. Really, just some way to let players know where they died.

**How the feature is useful**
As the owner of an smp, players often ask me to tell them their death coords. It would make it easier on me if players could find the coords on their own.

### PR DESCRIPTION
**Pull request by darbyjack**
Requested via https://github.com/EssentialsX/Essentials/issues/2896

Closes #2896

### CODE REVIEW

**Review by Ichbinjoe — APPROVED** [review]
but why tho

**Comment by Ichbinjoe** [comment]
#2896 because I have to do all the work here

**Review by pop4959 — CHANGES_REQUESTED** [review]
This fixes the problem perfectly as per the issue mentioned. However, I'd highly suggest relabelling it as a general custom death message feature. In the future I could see server owners asking for extensions of this, and the current name of "send-coords-on-death" might not age well with new features / placeholders added, etc.

File: Essentials/src/messages.properties [hunk 1]
```diff
@@ -73,6 +73,7 @@ confirmPayment=\u00a77To \u00a7lCONFIRM\u00a77 payment of \u00a76{0}\u00a77, ple
 connectedPlayers=\u00a76Connected players\u00a7r
 connectionFailed=Failed to open connection.
 cooldownWithMessage=\u00a74Cooldown\: {0}
+coordsAfterDeath=\u00a76Your location of death was X:{0}, Y:{1}, Z:{2}
```
Comment (mdcfe) [inline Essentials/src/messages.properties:76]:
```suggestion
coordsAfterDeath=\u00a76You died in \u00a7e{0} \u00a76at \u00a7e{1}, {2}, {3}\u00a76.
```

We should provide the option to include the world name. I'm also not a fan of this presentation - specifying X: Y: Z: is redundant when almost all coordinates in Minecraft are XYZ and it's unlikely you'd mistake a Y and Z coordinate.

File: Essentials/src/config.yml [hunk 2]
```diff
@@ -430,6 +430,9 @@ afk-list-name: "none"
 # You can disable the death messages of Minecraft here.
 death-messages: true
 
+# Would you like to tell the player the coords they were at when they died?
```
Comment (mdcfe) [inline Essentials/src/config.yml:433]:
```suggestion
# When players die, should they receive the coordinates they died at?
```

**Review by mdcfe — CHANGES_REQUESTED** [review]
> However, I'd highly suggest relabelling it as a general custom death message feature.

I agree with @pop4959 that this could be labelled as a custom death message feature, but on the other hand server owners are free to modify this message if they want to anyway.

File: Essentials/src/messages.properties (hunk 1 above)
Comment (pop4959) [reply Essentials/src/messages.properties:76]: I was a bit iffy on it as well but I'd agree with x, y, z more as well, like how minecraft:tp and other built-ins display it:
image

**Comment by pop4959** [comment]
> > However, I'd highly suggest relabelling it as a general custom death message feature.
> 
> I agree with @pop4959 that this could be labelled as a custom death message feature, but on the other hand server owners are free to modify this message if they want to anyway.

That's true, but wouldn't you want to relabel it now while we still can easily do so? I feel like changing the name of the feature later causes unnecessary headaches, for both us and server owners. I'm just saying this because from experience server owners tend to have a lot of different needs, so versatility is important. The original request only caters to a single niche use of sending a custom message on death (for coordinates). What if I want to include the death cause in the message? Or maybe (with eventual third party placeholders) the town or faction I was killed in?

I will re-iterate that as far as solving the original issue, this PR needs no changes. However the only change needed to future-proof this is to give it a better name and purpose (which should only take a few minutes to change). This would avoid awkwardness from getting requests for other send-X-after-death features in the future, or needing to refer people to change the "coordsAfterDeath" message when they want to send something other than coordinates after death, for example.

That may have been a long winded explanation but I wanted to give more context for my previous comment. Definitely a nitpick, but I think naming is an important subject to nitpick on.

**Comment by mdcfe** [comment]
Perhaps `send-info-after-death` and `infoAfterDeath` could be more appropriate. The main issue with leaving this possibility open for me is that adding more functionality will result in more and more unnamed placeholder, at least until custom placeholders are supported.

**Comment by pop4959** [comment]
Yep I was thinking about the same. Also, that's true, although I figure that's just sort of the nature of this kind of feature, at least until custom placeholder support exists. Realistically, I don't see there being a huge number of vanilla placeholders for this, but enough to justify generic-ifying things.

**Pull request closed without merging**